### PR TITLE
Implement shuffle

### PR DESCRIPTION
**Pull request by viluon**
Not sure if I did something wrong here and not sure if the example is portable. It works on CCEmuX, guessing from the dependency on CC libraries, I don't suppose luadash supports normal Lua environments.

Feel free to hate me if I did something wrong. This is an implementation of the [Fisher-Yates in-place shuffle](https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle), the same algorithm lodash uses, although I didn't have a look at their source.

Doesn't copy the metatable (although it looks like the other functions here don't do that either) and doesn't respect `__len`, which may be a problem.

### CODE REVIEW

File: library.lua
Comment (SquidDev) [inline library.lua]: Probably better to use `_.map(t, _.id)` - this is what `_.sort` does.

File: doc/shuffle.txt
Comment (SquidDev) [inline doc/shuffle.txt]: It's probably better to leave `math.randomseed` off, and provide an example value for when the seed is 0 - this is what sample does and what the test runner assumes.

**Review by SquidDev — CHANGES_REQUESTED** [review]
I'm not a maintainer, so feel free to ignore these :).

File: library.lua
Comment (Lignum) [reply library.lua]: Sounds like we need `_.clone`.

**Review by SquidDev — APPROVED** [review]

File: library.lua
Comment (Lignum) [inline library.lua]: I think we should wait until #19 is merged, so that we can replace this line.

**Pull request merged**
